include/ginkgo/core/base/executor.hpp

```diff
@@ -659,6 +659,17 @@ class Executor : public log::EnableLogging<Executor> {
         this->run(op);
     }
 
+    template <typename ClosureOmp, typename ClosureCuda, typename ClosureHip,
+              typename ClosureDpcpp>
+    void run(std::string name, const ClosureOmp& op_omp,
```

Comment:
Adding a new overload, would it make sense to extend this to include `ReferenceExecutor`? We can't change the other one, since it would break interface

Reply:
Any ordering preference? I would add it before `op_omp`, since I think we usually order them that way. But of course, this is then not compatible with the other overload.

Reply:
Sounds sensible, since this is a new overload. We can't add `reference` to the other one though, since that would create ambiguities with the std::string parameter and the additional templated parameter.

Reply:
I guess we could think about deprecating the other overload. Adding a name to an operation seems quite useful, especially since all ginkgo kernels are already named.